rcl_yaml_param_parser/src/parser.c

```diff
@@ -187,7 +187,12 @@ static rcl_ret_t add_name_to_ns(
       ns_len = strlen(cur_ns);
       name_len = strlen(name);
       sep_len = strlen(sep_str);
-      tot_len = ns_len + name_len + sep_len + 1U;
+      if (ns_len == sep_len) {
```

Comment:
self review: would be better to check the contents of the current ns, and skip appending if it ends with the separator already